doc/specs/stdlib_hash_procedures.md

```diff
@@ -0,0 +1,1748 @@
+It is a *pure* function for integer arrays, and an *elemental*
+function for character strings. 
+
+
+##### Example
+
+```fortran
+program demo_fnv_1_hash
+  use stdlib_32_bit_hash_codes, only: fnv_1_hash
+  use iso_fortran_env, only: int32
+  implicit none
+  integer(int32) :: hash
+  hash = fnv_1_hash([ 5, 4, 3, 1, 10, 4, 9])
+  print *, hash
+end program demo_fnv_1_hash
+```
+
+
+#### `FNV_1A_HASH`- calculates a hash code from a key
+
+##### Status
+
+Experimental
+
+##### Description
+
+Calculates a 32 bit hash code from a rank-1 integer array or a default
+character string.
+
+##### Syntax
+
+`code = [[stdlib_32_bit_hash_codes:fnv_1a_hash]]( key )`
+
+##### Class
+
+Pure/elemental function
+
+##### Argument
+
+`key`: Shall be a deferred length default character scalar expression
+or a rank-1 integer array expression of kind `INT8`, `INT16`,
+`INT32`, or `INT64`.
+It is an `intent(in)` argument.
+
+##### Result
+
+The result is a scalar integer of kind `INT32`.
+
+##### Note
+
+`FNV_1A_HASH` is an implementation of the alternative FNV-1a hash code of
+Glenn Fowler, Landon Curt Noll, and Phong Vo.
+It differs from typical implementations in that it also encodes the
+size of the structure in the hash code.
+This code is relatively fast on short keys, and is small enough that it
+will often be retained in the instruction cache if hashing is
+intermittent.
+As a result it should give good performance for typical hash table
+applications.
+This code does not pass any of the SMHasher tests, but the resulting
+degradation in performance due to its larger number of collisions is
+expected to be minor compared to its faster hashing rate.
+It is a *pure* function for integer arrays, and an *elemental*
+function for character strings. 
+
+##### Example
+
+```fortran
+program demo_fnv_1a_hash
+  use stdlib_32_bit_hash_codes, only: fnv_1a_hash
+  use iso_fortran_env, only: int32
+  implicit none
+  integer(int32) :: hash
+  hash = fnv_1a_hash( [ 5, 4, 3, 1, 10, 4, 9] )
+  print *, hash
+end program demo_fnv_1a_hash
+```
+
+
+#### `NEW_NMHASH32_SEED`- returns a valid input seed for `NMHASH32`
+
+##### Status
+
+Experimental
+
+##### Description
+
+Calculates a 32 bit "random" integer that is believed to be a valid
+seed for `NMHASH32` and is also different from the input seed.
+
+##### Syntax
+
+`code = call [[stdlib_32_bit_hash_codes:new_nmhash32_seed]]( seed )`
+
+##### Class
+
+Subroutine
+
+##### Argument
+
+`seed`: shall be a defined integer scalar variable of kind `INT32`.
+It is an `intent(inout)` argument. On input `seed` should be defined,
+and on output it will be different from the input `seed`.
+
+##### Note
+
+Currently there are no known bad seeds for `NMHASH32`, but if any are
+identified the procedure will be revised so that they cannot be
+returned. This subroutine uses Fortran's intrinsic
+ `RANDOM_NUMBER` and the values returned can be changed by calling the
+ intrinsic `RANDOM_INIT`.
```

Comment:
Isn't the intrinsic name `random_seed`? Also, stdlib_random provides a random number generator as well, is there a benefit to maybe prefer the stdlib version over the standard version?

Reply:
Since both modules are still experimental, I actually prefer to depend on an intrinsic, instead of a stdlib procedure for which the API might still change.

There is also an intrinsic `random_seed`, to restart or query the state of the intrinsic `random_number`